daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -528,96 +580,100 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
       readAs: Set[Ref.Party],
       commands: List[command.Command],
       optLocation: Option[Location],
-  )(implicit
-      ec: ExecutionContext
-  ): Future[Either[StatusRuntimeException, Seq[ScriptLedgerClient.CommandResult]]] = Future {
-    // Clear state at the beginning like in SBSBeginCommit for scenarios.
-    machine.returnValue = null
-    onLedger.commitLocation = optLocation
-    onLedger.localContracts = Map.empty
-    onLedger.globalDiscriminators = Set.empty
-    val speedyCommands = preprocessor.unsafePreprocessCommands(commands.to(ImmArray))._1
-    val translated = compiledPackages.compiler.unsafeCompile(speedyCommands)
-    machine.setExpressionToEvaluate(SEApp(translated, Array(SEValue.Token)))
-    onLedger.committers = actAs.toList.toSet
-    var result: Seq[ScriptLedgerClient.CommandResult] = null
-    while (result == null) {
-      machine.run() match {
-        case SResultNeedContract(coid, tid @ _, committers @ _, cbMissing, cbPresent) =>
-          scenarioRunner.lookupContract(coid, actAs.toSet, readAs, cbMissing, cbPresent).toTry.get
-        case SResultNeedKey(keyWithMaintainers, committers @ _, cb) =>
-          scenarioRunner.lookupKey(keyWithMaintainers.globalKey, actAs.toSet, readAs, cb).toTry.get
-        case SResultFinalValue(SUnit) =>
-          onLedger.ptx.finish match {
-            case PartialTransaction.CompleteTransaction(tx) =>
-              val results: ImmArray[ScriptLedgerClient.CommandResult] = tx.roots.map { n =>
-                tx.nodes(n) match {
-                  case create: NodeCreate[ContractId] =>
-                    ScriptLedgerClient.CreateResult(create.coid)
-                  case exercise: NodeExercises[_, ContractId] =>
-                    ScriptLedgerClient.ExerciseResult(
-                      exercise.templateId,
-                      exercise.choiceId,
-                      exercise.exerciseResult.get,
-                    )
-                  case n =>
-                    // Root nodes can only be creates and exercises.
-                    throw new RuntimeException(s"Unexpected node: $n")
+  )(implicit ec: ExecutionContext): Future[
+    Either[StatusRuntimeException, (RichTransaction, Seq[ScriptLedgerClient.CommandResult])]
+  ] =
```

Comment:
I recommend reviewing this with whitespace changes hidden. Unfortunately, scalafmt is pretty dumb here.